src/tests/public_api/test_helpers/proxy.h

```diff
@@ -119,9 +121,129 @@ class Proxy : public core::Thread, private packet::IWriter {
               == 0);
     }
 
+    Proxy(const roc_endpoint* receiver_source_endp,
+          const roc_endpoint* receiver_repair_endp,
+          const roc_endpoint* receiver_control_endp,
+          size_t n_source_packets,
+          size_t n_repair_packets,
+          size_t n_control_packets,
+          unsigned flags)
+        : packet_pool_("proxy_packet_pool", arena_)
+        , buffer_pool_("proxy_buffer_pool", arena_, 2000)
+        , queue_(packet::ConcurrentQueue::Blocking)
+        , net_loop_(packet_pool_, buffer_pool_, arena_)
```

Comment:
It's not a good idea to copy-paste this huge constructor. If we need two constructors, we'd need to extract common code into a method and reuse it.

However, in this case I think it's better to have just one constructor and pass NULL for control endpoint in tests that don't use it. It's better to make it explicit.

I saw your message about backwards compatibility but didn't have a chance to answer before you opened PR, sorry. We don't need to preserve compatibility here and want to update tests instead.